EventFilter/L1TRawToDigi/plugins/OmtfUnpacker.cc

```diff
@@ -0,0 +1,876 @@
+                                    data.quality(),
+                                    data.wireGroup(),
+                                    data.halfStrip(),
+                                    data.clctPattern(),
+                                    data.bend(),
+                                    data.bxNum()+3) ;
+          LogTrace("") << digi << std::endl;
+          producedCscLctDigis->insertDigi( cscId, digi); 
+
+        } 
+
+        //
+        // OMTF (muon) data
+        //
+        if (DataWord64::omtf==recordType) {
+          OmtfDataWord64   data(*word);
+          LogTrace("") <<"OMTF->MUON " << std::endl;
+          LogTrace("") << data << std::endl;
+          l1t::tftype  overlap = (fedId==1380) ? l1t::tftype::omtf_neg :  l1t::tftype::omtf_pos;
+          unsigned int iProcessor = bh.getAMCNumber()/2;   //0-5 
+          l1t::RegionalMuonCand digi;
+          digi.setHwPt(data.pT());
+          digi.setHwEta(data.eta());
+          digi.setHwPhi(data.phi());
+          digi.setHwSign(data.ch());
+          digi.setHwSignValid(data.vch());
+          digi.setHwQual(data.quality());
+          std::map<int, int> trackAddr;
+          trackAddr[0]=data.layers();
+          trackAddr[1]=0;
+          trackAddr[2]=data.weight_lowBits();
+          digi.setTrackAddress(trackAddr);
+          digi.setTFIdentifiers(iProcessor, overlap);
+          int bx = data.bxNum()-3;
+          LogTrace("")  <<"OMTF Muon, BX="<<bx<<", hwPt="<<digi.hwPt()<< std::endl;
+          if(std::abs(bx) <= 3) producedMuonDigis->push_back(bx,digi);
+        }
+
+
+        //
+        // DT data
+        //
+        if (DataWord64::dt==recordType) {
+          DtDataWord64   data(*word);
+          LogTrace("") <<"OMTF->DT " << std::endl;
+          LogTrace("") << data << std::endl;
+          int bx = data.bxNum()-3;
+          int whNum = (fedId==1380) ? -2 : 2;
+          int sector =   (bh.getAMCNumber()/2)*2 + data.sector();
+          if (sector==12) sector=0;
+          int station =  data.station()+1;
+//  unsigned int valid() const { return valid_; }
+//  unsigned int fiber() const { return fiber_; }
+//          if(data.valid())
+           phi_Container.push_back( L1MuDTChambPhDigi( bx, whNum, sector, station,
+                                    data.phi(), data.phiB(), data.quality(), 
+                                    0,                // utag/Ts2Tag 
+                                    data.bcnt_st())); //ucnt/BxCnt  
+          int pos[7];
+          int posQual[7];
+          for (unsigned int i=0; i<7; i++) { pos[i] = (data.eta() >> i & 1); posQual[i] = (data.etaQuality() >> i & 1); }
+            if(data.eta())the_Container.push_back(L1MuDTChambThDigi(bx,whNum, sector, station, pos, posQual)); 
+        }
+
+      }
+    }         
+
+  } 
+  event.put(std::move(producedRPCDigis),"OmtfUnpack");
+  event.put(std::move(producedCscLctDigis),"OmtfUnpack");
+  event.put(std::move(producedMuonDigis),"OmtfUnpack"); 
```

Comment:
The module name should be sufficient to distinguish.
CSC LCT digis unpacked by the EMTF unpacker would come from the module emtfStage2Digis and the ones unpacked by the OMTF unpacker come from the omtfDigis module.
The EMTF unpacker is called emtfStage2Digis and the BMTF unpacker bmtfDigis. I do not know what the guideline is.